server/src/main/java/org/elasticsearch/common/util/CollectionUtils.java

```diff
@@ -317,6 +317,20 @@ public static int sortAndDedup(final BytesRefArray bytes, final int[] indices) {
         return list;
     }
 
+    /**
+     * Creates a copy of the given collection with the given element appended.
+     *
+     * @param collection collection to copy
+     * @param element    element to append
+     */
+    @SuppressWarnings("unchecked")
+    public static <E> List<E> appendToCopy(Collection<E> collection, E element) {
+        final int size = collection.size() + 1;
```

Comment:
I think this can be written more succinctly and without the unchecked warning?

```
return Stream.concat(collection.stream(), Stream.of(element)).collect(Collectors.toList())
```

Reply:
True, but it's slower and returns a mutable list and less efficiently sized list. The former might matter in some of the snapshot/blob-store usage I added where this is called in surprisingly hot code.